proof-systems/src/darlin/t_dlog_acc_marlin/mod.rs

```diff
@@ -650,79 +622,32 @@ where
         pc_vk_g2: &<PC<G2, FS> as PolynomialCommitment<G2>>::VerifierKey,
         index_vk_g1: &VerifierKey<G1, G2, FS>,
         index_vk_g2: &VerifierKey<G2, G1, FS>,
-        inner_sumcheck_acc: &DualSumcheckItem<G1, G2>,
-        dlog_acc: &DualDLogItem<G1, G2>,
+        acc: &DualTDLogItem<G1, G2, PC<G1, FS>, PC<G2, FS>>,
     ) -> Result<
         (
-            (
-                DensePolynomial<G1::ScalarField>,
-                DensePolynomial<G2::ScalarField>,
-            ),
-            (
-                DensePolynomial<G1::ScalarField>,
-                DensePolynomial<G2::ScalarField>,
-            ),
+            DensePolynomial<G2::ScalarField>,
+            DensePolynomial<G2::ScalarField>,
         ),
         Error<<PC<G1, FS> as PolynomialCommitment<G1>>::Error>,
     > {
-        let dlog_poly_g1 = Self::check_dlog_item(pc_vk_g1, &dlog_acc.0[0])?;
-        let inner_sumcheck_poly_g1 =
-            Self::check_inner_sumcheck_item(pc_vk_g1, index_vk_g1, &inner_sumcheck_acc.0)?;
-
-        let dlog_poly_g2 = Self::check_dlog_item(pc_vk_g2, &dlog_acc.1[0])?;
-        let inner_sumcheck_poly_g2 =
-            Self::check_inner_sumcheck_item(pc_vk_g2, index_vk_g2, &inner_sumcheck_acc.1)?;
-
-        let dlog_polys = (dlog_poly_g1, dlog_poly_g2);
-        let inner_sumcheck_polys = (inner_sumcheck_poly_g1, inner_sumcheck_poly_g2);
-
-        Ok((inner_sumcheck_polys, dlog_polys))
-    }
-
-    fn check_dlog_item<G: IPACurve>(
-        pc_vk: &<PC<G, FS> as PolynomialCommitment<G>>::VerifierKey,
-        dlog_item: &DLogItem<G>,
-    ) -> Result<DensePolynomial<G::ScalarField>, Error<<PC<G, FS> as PolynomialCommitment<G>>::Error>>
-    {
-        let time = start_timer!(|| "Checking dlog accumulator item");
-        let verifier_output = <PC<G, FS> as PolynomialCommitment<G>>::hard_verify(pc_vk, dlog_item)
-            .map_err(Error::from_pc_err)?;
+        let check_acc = DualTDLogAccumulator::<_, _, _, _, FS>::check_item(
+            &(
+                &(&(index_vk_g1, pc_vk_g1), pc_vk_g1),
+                &(&(index_vk_g2, pc_vk_g2), pc_vk_g2),
```

Comment:
As discussed offline, I don't bother the references but the duplication in the pc_vk

Reply:
No more duplication of pc_vk after the removal of the CompositeAccumulator struct in favor of the explicit implementation of TDLogAccumulator